### PR TITLE
Doc: Adding contribution guide for eos_cli_config_gen

### LINKED ISSUE #3286: Doc: Improve contribution guide
Improve the contribution guide with details like below. Taken from a discussion, so reword to fit as a document:

--------

So in general we try to avoid introducing some new knobs in `eos_cli_config_gen` and `eos_designs` in one go to make things clearer. Usually, we just recommend one PR to add the low-level CLI in `eos_cli_config_gen` and then a new one on top to do the `eos_designs` bit. Thus I will structure a list of tasks based on this assumed separation and tick the steps you have followed already.

Make sure to install the `requirements-dev.txt` to be able to complete the process (I believe you have already but mentioning it just in case)

## `eos_cli_config_gen`

This part of the change is aimed at adding some new knobs. The following tasks are required:
- [x] Adding the knob to the appropriate schema fragments file in the `roles/eos_cli_config_gen/schema/schema_fragments` directory (or create a new file if adding a top-level feature)
- [ ] Editing the Jinja2 templates in `roles/eos_cli_config_gen/templates/eos` and `roles/eos_cli_config_gen/templates/documentation` (or adding new template if adding a top-level feature, in that case, you should also modify the `roles/eos_cli_config_gen/templates/eos-device-documentation.j2` and ``roles/eos_cli_config_gen/templates/eos-intended-config.j2 to add these new templates, trying to respect the order in the EOS CLI)
  * __NOTE__: This one is partially done in your PR as you have modified only the `ethernet-interface.j2` template for config and not the `port-channel.j2` one where you have added the `l2mru` knob in the schema. Also for now you have not added anything to the documentation templates (it may or may not be necessary)
- [x] Run `pre-commit run schemas --all` to re-generate the eos_cli_config_gen schema with your modifications  (you need to do this every time you change the schema, even if this is just a description changed). This is also used to add the new options to our documentation.
- [ ] Add some molecule test in n the `molecule/eos_cli_config_gen` scenario exercising your new knob configuration
  * __NOTE__: In your case you would need to add to both `molecule/eos_cli_config_gen/inventory/host_vars/ethernet-interfaces.yml` and `molecule/eos_cli_config_gen/inventory/host_vars/port-channel-interfaces.yml` (re-using existing interfaces is fine as this is only for testing purposes)
 - [ ] Run the molecule test locally to generate the new expected configuration and documentation for these two hosts (we recommend running the full scenario using from `ansible_collections/arista/avd/molecule` folder the command `make  converge MOLECULE=eos_cli_config_gen`
 - [ ] Run all pre-commit checks `pre-commit run  --all`  to make sure that all files you have added are correct and won't be rejected by CI (this command will take ~1min)
 - [ ] Review that everything added looks correct and push to Github!

## `eos_designs`

> Is there anything to do in "ansible_collections/arista/avd/roles/eos_designs"... as l2_mru` is not part of a design I suppose not?

I just would like to point out that if you don't need the `l2_mru` in `eos_designs` you should remove the changes you have made to the python code there. Otherwise, there is a bit more work to do indeed. This can be done only after the `eos_cli_config_gen` part has been completed (hence why we usually recommend 2 PRs)

The process is very similar (for simple changes like yours) except for the jinja2 template part where you need to edit some Python code

- [ ] Adding the knob to the appropriate schema fragments file in the `roles/eos_designs/schema/schema_fragments` directory
  * __NOTE__:  For you it would be `eos_designs/schemas/schema_fragments/defs_adapter_config.schema.yml` - something like https://github.com/aristanetworks/ansible-avd/blob/devel/ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/defs_adapter_config.schema.yml#L44 .
- [x] Editing the relevant python_modules
  * __NOTE__:  This has been done for you but you now need to add the knob to the `eos_designs` schema for adapters or it will never be possible to use it
- [ ] Run `pre-commit run schemas --all`  to re-generate the eos_designs schema with your modifications  (you need to do this every time you change the schema, even if this is just a description changed). This is also used to add the new options to our documentation.
- [ ] Add some molecule test in n the `molecule/eos_designs_unit_tests` scenario exercising your new knob configuration
  * __NOTE__: In your case you would need to add the `l2mru` knob to some adapters, one for port-channel and one for ethertnet-interfaces. It may be simpler in your case to add to the `molecule/eos_designs_unit_tests/inventory/host_vars/connected_endpoints.yml` file
 - [ ] Run the molecule test locally to generate the new expected configuration and documentation for this two hosts (we recommend running the full scenario using from `ansible_collections/arista/avd/molecule` folder the command `make  converge MOLECULE=eos_designs_unit_tests`
 - [ ] Run all pre-commit checks `pre-commit run  --all`  to make sure that all files you have added are correct and won't be rejected by CI (this command will take ~1min)
 - [ ] Review that everything added looks correct and push to Github!

Let us know if you need further help

_Originally posted by @gmuloc in https://github.com/aristanetworks/ansible-avd/issues/3164#issuecomment-1735054320_

### PR DESCRIPTION
**Pull request by Shivani-gslab**
## Related Issue(s)

Fixes #3286 

## Component(s) name

`arista.avd.<role-name>`

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
+<!--
+  ~ Copyright (c) 2023-2024 Arista Networks, Inc.
+  ~ Use of this source code is governed by the Apache License 2.0
+  ~ that can be found in the LICENSE file.
+  -->
+
+# Contribution Guide for `eos_cli_config_gen` role
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md]:
```suggestion
# Contribution Guide for the `eos_cli_config_gen` role
```

File: ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+This document outlines the steps and checklist for contributing to the `eos_cli_config_gen` role under Arista AVD.
+
+## Steps to add a new feature to eos_cli_config_gen role
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md]:
```suggestion
## Steps to add a new feature to the `eos_cli_config_gen` role
```

File: ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+1. **Primary Key Placement:** For list-type data-models, place primary keys at the top, for readability.
+2. **Key Naming:**
+    - Follow EOS CLI for key names, when creating new schema keys.
+    - Use plural for keys that represent multiple elements (e.g., sample_policies).
+3. **Descriptions:**
+    - Only add descriptions to the keys when they provide additional context beyond the key name.
+    - Refer Arista documentation for description content.
+    - Ensure all descriptions end with punctuation.
+    - Highlight the key names in description, like - `<key_name>`.
+4. **Type Conversion:** Add `convert_types: [str]` for `type: int` keys.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md]: And convert_types: [int] for type: str if it can be numeric

File: ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+5. **Defaults:** Avoid using `defaults` in eos_cli_config_gen.
+6. **Min/Max:** Specify min/max values in the schema if they are defined in the EOS CLI.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md]:
```suggestion
6. **Min/Max:** Specify min/max values in the schema if they are defined in the EOS CLI. Make sure to check on different hardware platforms.
```

File: ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+2. **Key Naming:**
+    - Follow EOS CLI for key names, when creating new schema keys.
+    - Use plural for keys that represent multiple elements (e.g., sample_policies).
+3. **Descriptions:**
+    - Only add descriptions to the keys when they provide additional context beyond the key name.
+    - Refer Arista documentation for description content.
+    - Ensure all descriptions end with punctuation.
+    - Highlight the key names in description, like - `<key_name>`.
+4. **Type Conversion:** Add `convert_types: [str]` for `type: int` keys.
+5. **Defaults:** Avoid using `defaults` in eos_cli_config_gen.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md]:
```suggestion
5. **Defaults:** Avoid using `default` in eos_cli_config_gen.
```

File: ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+6. **Min/Max:** Specify min/max values in the schema if they are defined in the EOS CLI.
+
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md:51]: Add something about valid values

File: ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+### Creating Jinja2 Templates
+
+Edit the appropriate jinja2 templates in `pyavd/_eos_cli_config_gen/j2templates/eos` and `pyavd/_eos_cli_config_gen/j2templates/documentation` to generate the desired configuration and documentation.
+
+Add new jinja2 template if adding a top-level feature, also modify the `pyavd/_eos_cli_config_gen/j2templates/eos-intended-config.j2` and `pyavd/_eos_cli_config_gen/j2templates/eos-device-documentation.j2` to add these new templates, trying to respect the order in the EOS CLI.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md]:
```suggestion
Add a new jinja2 template if adding a top-level feature, also modify the `pyavd/_eos_cli_config_gen/j2templates/eos-intended-config.j2` and `pyavd/_eos_cli_config_gen/j2templates/eos-device-documentation.j2` to add these new templates, trying to respect the order in the EOS CLI.
```

File: ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+
+#### Jinja2 Templates Guidelines
+
+1. **Code Indentation:** Keep less indented code to improve readability.
+2. **Variable Naming:** Use meaningful variable names.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md]:
```suggestion
2. **Variable Naming:** Use meaningful variable names. Avoid short variables like `ei` for `ethernet_interface`
```

File: ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+4. **Natural Sorting:** Use `arista.avd.natural_sort` for sorting the `for loops` after checking on EOS CLI.
+5. **Defined Checks:**
+    - Avoid `arista.avd.defined` check for parent keys when directly checking for child keys.
+    - Avoid `arista.avd.defined` check for primary and required keys.
+    - Avoid`arista.avd.defined` check when using `arista.avd.default()` and `arista.avd.natural_sort` filters.
+6. **Password Security:** Avoid displaying passwords in the documentation template and use the `arista.avd.hide_passwords` filter to hide it.
+7. **Config Order:** Ensure the order and indentation of configuration matches EOS CLI.
+8. **Exclamation Marks:** Place exclamation marks `!` correctly as per the EOS running-config.
+9. Along with EOS config template, update the documentation template as well (if required).
+10. Implement only commands visible in running-config of the EOS device.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md]:
```suggestion
10. Implement only commands visible in `show running-config all` of the EOS device. We should not hide config if given by the user.
```

File: ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+### Build Schemas and Documentation
+
+Run `pre-commit run schemas --all` to re-generate the eos_cli_config_gen schema with any modifications. This command should be executed every time the schema is changed, even if only a description is updated.
+It also updates the documentation with new options.
+
+### Add Molecule Tests
+
+Add some molecule tests in the `ansible_collections/arista/avd/molecule/eos_cli_config_gen` scenario exercising the new knob configuration.
+
+When marking any key as "deprecated," move the related tests to the `eos_cli_config_gen_deprecated_vars` molecule scenario and add any missing tests if necessary.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md]:
```suggestion
When marking any key as "deprecated", move the related tests to the `eos_cli_config_gen_deprecated_vars` molecule scenario and add any missing tests if necessary.
```

File: ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+Check the PyAVD test coverage report by running `tox -e coverage,report` and work on improving the coverage where possible.
+
+### Update Documentation
+
+If the proposed feature requires any changes to the documentation, make sure to update it accordingly.
+If there are any breaking changes introduced, document them in the porting guide.
+
+### Run Pre-commit Checks
+
+Run all pre-commit checks `pre-commit run --all` to ensure that all files added or modified are correctly following the coding standards and formatting rules.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md]:
```suggestion
Run all pre-commit checks `pre-commit run` to ensure that all files added or modified are correctly following the coding standards and formatting rules.
```

Comment (Shivani-gslab) [reply ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md]: `pre-commit run` checks staged files only

File: ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md (hunk 3 above)
Comment (Shivani-gslab) [reply ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md]: Added

File: ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md (hunk 6 above)
Comment (Shivani-gslab) [reply ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md:51]: Added

Comment (MaheshGSLAB) [reply ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md:51]: Lets add the the use of required key

File: ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md [hunk 12]
```diff
@@ -0,0 +1,111 @@
...
+Before pushing the changes to GitHub, carefully review all the modifications.
+Confirm that all new features work as intended and that existing features are unaffected. Once satisfied, proceed to push the changes to the repository.
+
+## Checklist to review an eos_cli_config_gen PR
+
+1. Check that all the CI checks are passing.
+2. If the PR addresses an issue raised in the repository, confirm that the issue is fully resolved by the PR.
+3. Refer to the Arista documentation for a deeper understanding of the proposed feature.
+4. Verify that the schema adheres to EOS CLI and all relevant guidelines mentioned above.
+5. Ensure that the min/max values are specified in the schema if they are defined in the EOS CLI.
```
Comment (MaheshGSLAB) [inline ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md]: add for valid_values and required key

File: ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md (hunk 11 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md]: right. But pre-commit run --all takes forever :)

File: ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md (hunk 6 above)
Comment (Shivani-gslab) [reply ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md:51]: Added!

File: ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md (hunk 12 above)
Comment (Shivani-gslab) [reply ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md]: Added

File: ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md (hunk 6 above)
Comment (Shivani-gslab) [reply ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md:51]: Removed it, as suggested by claus on other PRs.

**Review by MaheshGSLAB — DISMISSED** [review]
LGTM

File: ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md [hunk 13]
```diff
@@ -0,0 +1,111 @@
...
+
+# Contribution Guide for the `eos_cli_config_gen` role
+
+This document outlines the steps and checklist for contributing to the `eos_cli_config_gen` role under Arista AVD.
+
+## Steps to add a new feature to the `eos_cli_config_gen` role
+
+### Prepare development environment
+
+Follow the [Development Tooling Guide](https://avd.arista.com/5.0/docs/contribution/development-tooling.html).
```
Comment (gmuloc) [inline ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md]: So two things
1. use stable
2. I would think you could use a relative link there instead of absolute

```suggestion
Follow the [Development Tooling Guide](https://avd.arista.com/stable/docs/contribution/development-tooling.html).
```

same for all other links

File: ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md [hunk 14]
```diff
@@ -0,0 +1,111 @@
...
+
+### Schema creation
+
+Add the schema for new feature as per EOS CLI to the appropriate schema fragments file in the `pyavd/_eos_cli_config_gen/schema/schema_fragments` directory or create a new schema file if adding a top-level feature.
```
Comment (gmuloc) [inline ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md]: I would start from the root
```suggestion
Add the schema for new feature as per EOS CLI to the appropriate schema fragments file in the `python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments` directory or create a new schema file if adding a top-level feature.
```

File: ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md [hunk 15]
```diff
@@ -0,0 +1,111 @@
...
+
+This document outlines the steps and checklist for contributing to the `eos_cli_config_gen` role under Arista AVD.
+
+## Steps to add a new feature to the `eos_cli_config_gen` role
+
+### Prepare development environment
+
+Follow the [Development Tooling Guide](https://avd.arista.com/5.0/docs/contribution/development-tooling.html).
+
+### Schema creation
```
Comment (gmuloc) [inline ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md:27]: I would add an overview of whats needed as a list before this like

* schema
* templates (eos and documentation)
* molecule (tests)

File: ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md [hunk 16]
```diff
@@ -0,0 +1,111 @@
...
+4. **Type Conversion:**
+    - Add `convert_types: [str]` for `type: int` keys.
+    - Add `convert_types: [int]` for `type: str` if it can be numeric.
+5. **Defaults:** Avoid using `default` in eos_cli_config_gen.
+6. **Min/Max:** Specify min/max values in the schema if they are defined in the EOS CLI. Make sure to check on different hardware platforms.
+7. **Valid_Values:** Specify valid options in the schema as per the EOS CLI and ensure compatibility across different hardware platforms.
+
+### Creating Jinja2 Templates
+
+Edit the appropriate jinja2 templates in `pyavd/_eos_cli_config_gen/j2templates/eos` and `pyavd/_eos_cli_config_gen/j2templates/documentation` to generate the desired configuration and documentation.
```
Comment (gmuloc) [inline ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md]:
```suggestion
Edit the appropriate Jinja2 templates in `pyavd/_eos_cli_config_gen/j2templates/eos` and `pyavd/_eos_cli_config_gen/j2templates/documentation` to generate the desired configuration and documentation.
```

File: ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md [hunk 17]
```diff
@@ -0,0 +1,111 @@
...
+
+Add a new jinja2 template if adding a top-level feature, also modify the `pyavd/_eos_cli_config_gen/j2templates/eos-intended-config.j2` and `pyavd/_eos_cli_config_gen/j2templates/eos-device-documentation.j2` to add these new templates, trying to respect the order in the EOS CLI.
```
Comment (gmuloc) [inline ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md]:
```suggestion
When adding a top-level feature, add a new jinja2 template following the naming convention and modify the `pyavd/_eos_cli_config_gen/j2templates/eos-intended-config.j2` and `pyavd/_eos_cli_config_gen/j2templates/eos-device-documentation.j2` to add these new templates where relevant, in particular to respect EOS CLI order.
```

File: ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md [hunk 18]
```diff
@@ -0,0 +1,111 @@
...
+6. **Password Security:** Avoid displaying passwords in the documentation template and use the `arista.avd.hide_passwords` filter to hide it.
+7. **Config Order:** Ensure the order and indentation of configuration matches EOS CLI.
+8. **Exclamation Marks:** Place exclamation marks `!` correctly as per the EOS running-config.
+9. Along with EOS config template, update the documentation template as well (if required).
+10. Implement only commands visible in `show running-config all` of the EOS device. We should not hide config if given by the user.
+11. Validate the template using j2lint tool, run `pre-commit run j2lint --all`.
+
+### Build Schemas and Documentation
+
+Run `pre-commit run schemas --all` to re-generate the eos_cli_config_gen schema with any modifications. This command should be executed every time the schema is changed, even if only a description is updated.
```
Comment (gmuloc) [inline ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md]: Starting AVD 5.1 - it will happen automaticaly when running molecule but will be required if trying with an external repo

File: ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md [hunk 19]
```diff
@@ -0,0 +1,111 @@
...
+It also updates the documentation with new options.
+
+### Add Molecule Tests
+
+Add some molecule tests in the `ansible_collections/arista/avd/molecule/eos_cli_config_gen` scenario exercising the new knob configuration.
```
Comment (gmuloc) [inline ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md]: Explain how to add to `hostX/xxx.yml` and adding to subsequent hosts to test extra combination

File: ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md [hunk 20]
```diff
@@ -0,0 +1,111 @@
...
+
+When marking any key as "deprecated", move the related tests to the `eos_cli_config_gen_deprecated_vars` molecule scenario and add any missing tests if necessary.
+
+### Run Molecule Tests
+
+Run `molecule converge -s eos_cli_config_gen` from the path `ansible_collections/arista/avd/` to execute the molecule tests locally and generate the new expected configuration and documentation for newly added test-cases.
+
+Check the PyAVD test coverage report by running `tox -e coverage,report` and work on improving the coverage where possible.
```
Comment (gmuloc) [inline ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md]: This could be hard to follow given the templates are compiled and we are not explaining how to work from there (which is not straighforward)

File: ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md [hunk 21]
```diff
@@ -0,0 +1,111 @@
...
+
+### Update Documentation
+
+If the proposed feature requires any changes to the documentation, make sure to update it accordingly.
+If there are any breaking changes introduced, document them in the porting guide.
```
Comment (gmuloc) [inline ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md]: @ClausHolbechArista / @carlbuchmann - porting guide for major releases /  release-notes for minor and maintenance releases?

File: ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md (hunk 11 above)
Comment (gmuloc) [reply ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md]: I see it still reads `--all` is it expected? (conversation was resolved)

File: ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md [hunk 22]
```diff
@@ -0,0 +1,111 @@
...
+## Checklist to review an eos_cli_config_gen PR
+
+1. Check that all the CI checks are passing.
+2. If the PR addresses an issue raised in the repository, confirm that the issue is fully resolved by the PR.
+3. Refer to the Arista documentation for a deeper understanding of the proposed feature.
+4. Verify that the schema adheres to EOS CLI and all relevant guidelines mentioned above.
+5. Ensure that the min/max and valid-values are specified in the schema if they are defined in the EOS CLI.
+6. Ensure that Jinja2 templates follow all the guidelines mentioned above.
+7. Check that the template generates valid configuration and documentation, maintaining the same configuration order and indentation as EOS CLI.
+8. Check out the PR in a local IDE and run all tests to ensure functionality.
```
Comment (gmuloc) [inline ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md]:
```suggestion
8. Check out the PR in a local IDE using the instructions in the PR comment and run all tests to ensure functionality.
```

File: ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md [hunk 23]
```diff
@@ -0,0 +1,111 @@
...
+9. Test the generated configuration through EAPI or CVP in the ATD lab.
```
Comment (gmuloc) [inline ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md]: not everybody may have an ATD lab - you can state `in a relevant lab (if availbale otherwise indicate you could not verify in a lab in your review comment.)`
```suggestion
9. Test the generated configuration through EAPI or CVP in the ATD lab.
```

File: ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md [hunk 24]
```diff
@@ -0,0 +1,111 @@
...
+10. If providing code suggestions, test them locally to ensure that your proposals work as intended.
```
Comment (gmuloc) [inline ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md]: Add something to state to add `inline suggestions whenever possible`

**Review by gmuloc — CHANGES_REQUESTED** [review]
I would add some general notes in some places stating

1. For PR writer to tag maintainers with questions even once they have pushed and to use draft PRs when things are not ready yet. (We can also discuss in the issue if requried)
2. For PR reviewer - to tag maintainers when they have questions

I think you are missing to introduce the new md file in the TOC for mkdocs (you need tp modify the mkdocs.yml)

File: ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md (hunk 19 above)
Comment (Shivani-gslab) [reply ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md]: Fixed

File: ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md (hunk 23 above)
Comment (Shivani-gslab) [reply ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md]: Updated

File: ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md (hunk 24 above)
Comment (Shivani-gslab) [reply ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md]: added

File: ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md (hunk 15 above)
Comment (Shivani-gslab) [reply ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md:27]: Added a list of steps

File: ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md (hunk 18 above)
Comment (gmuloc) [reply ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md]: basically if you run from source (which would happen in molecule / if you run the collection from its github repo) you would not need to do this when running through Ansible. When using pyavd you need to.

I think it is fair to state something like "When developing a feature you should be running from source using Ansible (either via molecule or from a test repo) and so the schemas and templates should be auto-recompiled during the "Verify Requirements" step as per https://avd.arista.com/stable/docs/contribution/development-tooling.html#running-from-source. When running pyavd or if you need to recompile for another reason use `pre-commit run schemas --all` and / or `pre-commit run templates --all`."

Comment (Shivani-gslab) [reply ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md]: Updated!

File: mkdocs.yml [hunk 25]
```diff
@@ -192,6 +192,7 @@ nav:
   - External Resources:
       - Ansible Collection User Guide: https://docs.ansible.com/ansible/latest/user_guide/collections_using.html
       - Ansible User Guide: https://docs.ansible.com/ansible/latest/user_guide/index.html
+      - Authoring eos_cli_config_gen Role Guide: docs/contribution/authoring_eos_cli_config_gen.md
```
Comment (carlbuchmann) [inline mkdocs.yml]: This should be in the `Contributing to AVD:` section

Comment (Shivani-gslab) [inline mkdocs.yml]:
```suggestion
```

File: mkdocs.yml [hunk 26]
```diff
@@ -192,6 +192,7 @@ nav:
   - External Resources:
```
Comment (Shivani-gslab) [inline mkdocs.yml:193]:
```suggestion
      - Authoring eos_cli_config_gen Role Guide: docs/contribution/authoring_eos_cli_config_gen.md
  - External Resources:
```

File: mkdocs.yml (hunk 25 above)
Comment (Shivani-gslab) [reply mkdocs.yml]: Thanks, fixed!

File: ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md (hunk 11 above)
Comment (Shivani-gslab) [reply ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md]: Since it was resolved, I didn't update it, assuming it's fine to keep it as it is.

File: ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md (hunk 6 above)
Comment (gmuloc) [reply ansible_collections/arista/avd/docs/contribution/authoring_eos_cli_config_gen.md:51]: added it back now that it works ;)

**Review by carlbuchmann — APPROVED** [review]
Great PR, LGTM!

**Review by gmuloc — APPROVED** [review]
Thanks @Shivani-gslab for this work! very helpful for contributor 🚢

**Review by ClausHolbechArista — APPROVED** [review]
